models/classes/search/tasks/UpdateResourceInIndex.php

```diff
@@ -57,7 +63,36 @@ public function __invoke($params): Report
             throw new common_exception_MissingParameter();
         }
 
-        $numberOfIndexed = $this->getSearchProxy()->index($this->getIndexDocuments($params));
+        $resourceUris = is_array($params[0]) ? $params[0] : [$params[0]];
+
+        return $this->getReport(
+            $this->getSearchProxy()->index(
+                $this->getIndexDocuments($resourceUris)
+            )
+        );
+    }
+
+    /**
+     * @throws common_exception_InconsistentData
+     * @throws common_Exception
+     */
+    private function getIndexDocuments(array $resourceUris): array
+    {
+        $documents = [];
+
+        foreach ($resourceUris as $resourceUri) {
+            $this->resourceUris[] = $resourceUri;
+
+            $documents[] = $this->getDocumentBuilder()->createDocumentFromResource(
+                $this->getResource($resourceUri)
+            );
+        }
+
+        return $documents;
```

Comment:
I know I wrote this code, but I should have optimized it :) 

```suggestion
        $documentBuilder = $this->getDocumentBuilder();
        $documents = [];

        foreach ($resourceUris as $resourceUri) {
            $this->resourceUris[] = $resourceUri;

            $documents[] = $documentBuilder->createDocumentFromResource($this->getResource($resourceUri));
        }

        return $documents;
```